.github/workflows/e2e-matrix.yaml

```diff
@@ -18,6 +18,10 @@ on:
         required: true
       E2E_SUBSCRIPTION_ID:
         required: true
+
+permissions:
+  contents: read
```

Comment:
I felt like there was something where I needed the higher level workflows to have the perms that the lower level ones requested, but we can also just see if that's the actual case